src/pykoala/instruments/koala_ifu.py

```diff
@@ -208,14 +215,20 @@ def koala_rss(path_to_file):
     info['fib_dec'] = np.rad2deg(koala_header['DECCEN']) + koala_fibre_table.data['Delta_DEC'] / 3600
     info['airmass'] = airmass_from_header(koala_header)
     # Read RSS file into a PyKoala RSS object
-    rss = read_rss(path_to_file, wcs=koala_wcs,
-                   bad_fibres_list=bad_fibres_list,
+    rss = read_rss(path_to_file, 
+                   wcs=koala_wcs,                         
+                   bad_fibres_list=bad_fibres_list,      
                    intensity_axis=0,
                    variance_axis=1,
-                   header=koala_header,
+                   header=koala_header,                  
                    fibre_table=koala_fibre_table,
```

Comment:
All of this information is never stored in the RSS object. Actually, the function `read_rss` is deprecated and should disappear. Every *instrument*_rss method within the instruments module should returns an RSS.

If you would like to force the class to include further information (that won't be used anywhere else in the pipeline), you can first create the RSS within `koala_rss`, and then create new attributes via `rss.new_attribute = value`.

Reply:
That is what I actually did, I created my own `koalaRSS` function that uses `koala_rss` (this one still calls `read_rss`) and this function creates a new dictionary `rss.koala_info` where I add the parameters I want to keep, so far:

```
koala_info_rss = dict(first_good_wave_per_fibre = None,
                      last_good_wave_per_fibre = None,
                      list_fibres_all_good_values = None,
                      rss_object_name = None,
                      path_to_file = None,
                      n_wave = None,
                      n_spectra = None,
                      spaxel_size = None,
                      aaomega_arm = None,
                      aaomega_grating = None,
                      KOALA_fov = None,
                      pos_angle = None,
                      RA_cen = None,
                      DEC_cen = None,
                      description = None,
                      valid_wave_min = None,
                      valid_wave_max = None,
                      valid_wave_min_index = None,
                      valid_wave_max_index = None
                      )
```
Somehow, I would have preferred to have at least some of these as `rss.item` as it is faster, but we agreed we will not add anything else to the rss except an extra `rss.info`, so that is what I did.

I also created `rss.koala_header` with all the info of the header and `rss.koala_wcs` (I'm quite sure I do not need this one, but just in case I'm keeping it for now).

Reply:
Please post the relevant code, but it doesn't look good. Anything that is specific to KOALA should be within `instruments/koala_ifu.py`, WEAVE-specific stuff should go into `instruments/weave.py`, etc.

By the way, shouldn't we call the instrument "koala" rather than "koala_ifu"?

Reply:
> By the way, shouldn't we call the instrument "koala" rather than "koala_ifu"?

>[!TIP]
> To rename a file in a git repository, you may use the web interface or
> `git mv <old-name> <new-name>`
> from the command line. By doing so, you inform `git` that it is indeed the same file. Otherwise, it will assume that you deleted the old one and created a completely new one, and its whole content will be copied (again) to the repo, wasting disk space.